shaptools/hdb_connector/connectors/pyhdb_connector.py

```diff
@@ -75,3 +77,39 @@ def disconnect(self):
         self._logger.info('disconnecting from SAP HANA database')
         self._connection.close()
         self._logger.info('disconnected successfully')
+
+    def isconnected(self):
+        """
+        Check the connection status. It checks if the socket is properly working
+
+        INFO: Sometimes the state is not changed unless a query is performed
+
+        Returns:
+            bool: True if connected False otherwise
+        """
+        if self._connection and self._connection.isconnected():
+            try:
+                self._connection._socket.getpeername()
```

Comment:
I wonder if this socket function is the best to get the socket status .

e.g https://docs.python.org/2/library/socket.html#socket.socket.getpeername
in some systems it is not supported.

Reply:
reading docs people refer to this https://docs.python.org/3/library/socket.html#socket.timeout

Reply:
I added before a comment but It was not saved properly, so here I go again...

The `timeout` is an Exception, so it is raised when some method trying to use the socket fails, and only works if the timeout is set. I don't think this is a better choice.

As far as I have checked, the `getpeername` does what we want. If the connection is dropped it fails. I have tested with many different options and this is the one that works best.
Otherwise, we should use a method to send or receive data, and I don't think that's a better option.

About the `On some systems this function is not supported` comment, In Suse family distros works fine, and I would bet that in most of the Unix systems.

I have not found a better option without over engineering so much, and besides that I have tested the functionality and works fine.